src/main/java/org/apache/fineract/api/BatchApi.java

```diff
@@ -40,7 +40,7 @@ public Batch batchDetails(@RequestParam String batchId, @RequestParam String req
     }
 
     @GetMapping("/batch/transactions")
-    public HashMap<String, String> batchTransactionDetails(@RequestParam String batchId) {
+    public HashMap<String, String> batchTransactionDetails(@RequestHeader("batchId") String batchId) {
```

Comment:
Batch id should be a request parameter and not a header.

Reply:
Unable to set request parameter in camel route of bulk processor. That's why using request header.